### PR TITLE
fix(config): add keemple power plug support

### PR DESCRIPTION
**Pull request by zollek**
Add support for keemple power plug.

I hope this time I don't have so many mistakes.

### CODE REVIEW

**Comment by zollek** [comment]
@zwave-js-bot fix lint

**Review by AlCalzone — CHANGES_REQUESTED** [review]
Please take a good look at our [style guide](https://zwave-js.github.io/node-zwave-js/#/config-files/style-guide) and apply it.

There are a bunch of things wrong here, e.g.
- unnecessary / too long descriptions
- Labels should be in **Title Case**
- Typos
- etc.

**Comment by knthm** [comment]
The Keemple KP-SO-02 is a branded Shenzhen Neo NAS-WR01ZE.

~~I suggest copying that [device's config](https://github.com/zwave-js/node-zwave-js/blob/master/packages/config/config/devices/0x0258/nas-wr01ze.json) and replacing the manufacturerId, productType, and productId with the Keemple-specific values.~~

I've a batch of unbranded NAS-WR01ZE's from a chinese distributor on which they accidentally applied the Keemple-brand firmware configuration.

**Update:** I think it's better to import the NAS-WR01ZE device configuration with the template engine and override the branding-specific values. See referenced PR below.

**Comment by zollek** [comment]
I've shortened it using Style guide and @KloudJack  advice.

Sorry for earlier, didn't know that such a guide exist. Simply copied the manual.

File: packages/config/config/devices/0x041a/keemple_power_plug.json [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+			"unit": "second",
+			"valueSize": 2,
+			"minValue": 30,
+			"maxValue": 32767,
+			"defaultValue": 300
+		},
+		{
+			"#": "8",
+			"label": "Maximum Current",
+			"description": "Maximum output current limit",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x041a/keemple_power_plug.json:144]: This description seems unnecessary

File: packages/config/config/devices/0x041a/keemple_power_plug.json [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+			"unit": "A",
+			"valueSize": 1,
+			"minValue": 1,
+			"maxValue": 16,
+			"defaultValue": 16
+		},
+		{
+			"#": "9",
+			"label": "Current Report Treshold",
+			"description": "Current treshold that triggers a report",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x041a/keemple_power_plug.json:154]: dito

File: packages/config/config/devices/0x041a/keemple_power_plug.json [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+			"options": [
+				{
+					"label": "Always off",
+					"value": 0
+				},
+				{
+					"label": "Previous state",
+					"value": 1
+				}
+			]
```
Comment (AlCalzone) [inline packages/config/config/devices/0x041a/keemple_power_plug.json:44]:
```suggestion
			"$import": "~/templates/master_template.json#state_after_power_failure_off_prev"
```

File: packages/config/config/devices/0x041a/keemple_power_plug.json [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+			"maxValue": 1,
+			"defaultValue": 1,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Disabled",
+					"value": 0
+				},
+				{
+					"label": "Active",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x041a/keemple_power_plug.json:60]:
```suggestion
					"label": "Enable",
```

File: packages/config/config/devices/0x041a/keemple_power_plug.json [hunk 5]
```diff
@@ -0,0 +1,177 @@
+{
+	"manufacturer": "keemple",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x041a/keemple_power_plug.json:2]:
```suggestion
	"manufacturer": "Keemple",
```

**Review by AlCalzone — CHANGES_REQUESTED** [review]
Thanks, this is better. Just a few minor changes.

Also, please rename the file to `kp-so-02.json`

**Comment by AlCalzone** [comment]
Oh, I just saw the other PR https://github.com/zwave-js/node-zwave-js/pull/4125/
Let me think about what's the better approach here.

**Comment by knthm** [comment]
> Oh, I just saw the other PR #4125 Let me think about what's the better approach here.

@AlCalzone any update on this?

If importing the whole device config feels too hackish, I could move most of the paramInformation into `~/0x0258/templates/shenzhen_neo_template.json` and import it from there (updating both Shenzhen and Keemple configs).

Almost half of the parameters are `base_enable_disable` which can be imported from the master_template as well.

This way both the NAS-WR01ZE and KP-SO-O2 configs would be structured as follows: https://github.com/zwave-js/node-zwave-js/blob/aff1bc6f1f431e250eb48bf344214a457b0ca290/packages/config/config/devices/0x0371/zw162.json#L2-L35

**Comment by AlCalzone** [comment]
Eh, sorry this slipped through the cracks. We've decided not to import the entire file - from our experience, it sometimes happens that OEM devices deviate from the original firmwares.

I think we have the most flexibility by having both import from a common template like you proposed. Could you do that?

**Comment by knthm** [comment]
@AlCalzone 

After a quick look at the Shenzhen Neo template and configs, I decided to cleanup the device configs: #4186

At first I'd also moved some of the power plug-specific parameters into the Shenzhen Neo manufacturer template, but after noticing that the NAS-WR01Z and NAS-WR01ZE both have functionally-identical parameters all with (annoyingly) different configurations, I removed them again as it felt out of place to have those in the manufacturer template when they only applied to one device from said manufacturer.

See the following parameter as an example:
https://github.com/zwave-js/node-zwave-js/blob/5827bc98fc9577a6586eb3fd58d62f1d38eb71f2/packages/config/config/devices/0x0258/nas-wr01z.json#L68-L77
https://github.com/zwave-js/node-zwave-js/blob/5827bc98fc9577a6586eb3fd58d62f1d38eb71f2/packages/config/config/devices/0x0258/nas-wr01ze.json#L146-L155

or

https://github.com/zwave-js/node-zwave-js/blob/5827bc98fc9577a6586eb3fd58d62f1d38eb71f2/packages/config/config/devices/0x0258/nas-wr01z.json#L48-L57
https://github.com/zwave-js/node-zwave-js/blob/5827bc98fc9577a6586eb3fd58d62f1d38eb71f2/packages/config/config/devices/0x0258/nas-wr01ze.json#L116-L125

:neutral_face:

I've updated #4125 which uses some master template parameters but no Shenzhen Neo template parameters.

**Pull request closed without merging**